src/server/plugins/override-response-type.ts

```diff
@@ -26,9 +26,30 @@ function handleBoom(response: Response, contentType: string) {
   }
 }
 
+function applyEmptyStatusCode(request: Request) {
+  const { method, response } = request
+  const { source, statusCode } = response
+
+  // slight customisations over the Hapi default behaviour
+  // see: https://github.com/hapijs/hapi/blob/b92664d9e4e1a184e7e2d958049af0f065d04f02/lib/response.js#L68
+
+  if (source !== null) {
+    return
+  }
+
+  if (statusCode === 204 ||
+    (statusCode === 200 && request.route.settings.response.emptyStatusCode === 204) ||
+    (method === 'options' && statusCode === 200)) {
```

Comment:
and then you could even deny the request if the method isn't `GET`, if you wanted to